plugins/inventory/digitalocean.py

```diff
@@ -0,0 +1,205 @@
+    prefix: ''
+    separator: ''
+compose:
+  ansible_host: do_networks.v4 | selectattr('type','eq','public')
+    | map(attribute='ip_address') | first
+  class: do_size.description | lower
+  distro: do_image.distribution | lower
+'''
+
+import json
+from ansible.errors import AnsibleParserError
+from ansible.module_utils.urls import Request
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'community.digitalocean.digitalocean'
+
+    def verify_file(self, path):
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(
+                    ('do_hosts.yaml', 'do_hosts.yml',
+                     'digitalocean.yaml', 'digitalocean.yml',
+                     'digital_ocean.yaml', 'digital_ocean.yml')
+            ):
+                valid = True
+            else:
+                self.display.vvv(
+                    'Skipping due to inventory source file name mismatch. '
+                    'The file name has to end with one of the following: '
+                    'do_hosts.yaml, do_hosts.yml '
+                    'digitalocean.yaml, digitalocean.yml, '
+                    'digital_ocean.yaml, digital_ocean.yml.')
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        # cache settings
+        self._read_config_data(path)
+        self.cache_key = self.get_cache_key(path)
+
+        self.use_cache = self.get_option('cache') and cache
+        self.update_cache = self.get_option('cache') and not cache
+
+        results = []
+        if not self.update_cache:
+            try:
+                results = self._cache[self.cache_key]['digitalocean']
+            except KeyError:
+                pass
+
+        if not results:
+            if self.cache_key not in self._cache:
+                self._cache[self.cache_key] = {'digitalocean': ''}
+
+        # request parameters
+        base_url = 'https://api.digitalocean.com/v2'
+        resource = 'droplets'
+        pagination = self.get_option('pagination')
+        url = '{0}/{1}?per_page={2}'.format(base_url, resource, pagination)
+
+        api_token = self.get_option('api_token')
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer {0}'.format(api_token)
+        }
+
+        # send request
+        req = Request(headers=headers)
+        try:
+            self.display.vvv('Sending request to {0}'.format(url))
+            response = json.load(req.get(url))
+            key = 'droplets'
+            payload = response.get(key)
+            if not payload:
+                raise AnsibleParserError(
+                    "There is no droplet information received")
```

Comment:
The API docs state that the `droplets` will be a list of maps, so this check is not really needed. If the DigitalOcean decides to break the compatibility, the kind of exception we get does not matter much, so we can just let the plugin error-out with a key error and be done with it.

```suggestion
            payload = response['droplets']
```

Reply:
ok, let it be like that. After examining the unit tests maybe I became a little bit paranoid.